### ISSUE TITLE
asyncContext on development is dead in nightly build

### ISSUE THREAD

**Issue by daniluk4000** [nuxt/nuxt#27319]
### Environment

- Operating System: Linux
- Node Version:     v20.12.2
- Nuxt Version:     3.12.0-28586253.17751795
- CLI Version:      3.11.2-1716413292.bfaf128
- Nitro Version:    2.10.0-28605228.943f11e0
- Package Manager:  yarn@4.2.2
- Builder:          -
- User Config:      extends, dir, nitro, alias, runtimeConfig, avModule, app, plugins, modules, yandexMaps
- Runtime Modules:  ~/modules/entrypoint.ts, @misc/base-vue-module/module@5.1.39, @nuxt/test-utils/module@3.13.1, vue-yandex-maps/nuxt
- Build Modules:    -

### Reproduction

Later, still debugging and trying to fix this

### Describe the bug

Caused by: https://github.com/nuxt/nuxt/commit/177517951c9c034758f4600d70856fc4649f37f6

Basically asyncContext has no effect now. Worked great on 3.12.0-28585351.68f4b193.

### Additional context

Sometimes I think I'm the only one using asyncContext in whole ecosystem

Happens in development environment only

**Comment by daniluk4000** [comment]
nvm, our mistake with getContext in some of plugins
